### PR TITLE
Adds an Instagib Rifle

### PR DESCRIPTION
**Pull request by MMMiracles**
rifles

A rifle based on the Unreal series 'instagib' mode, infinite ammo and gibs anything it manages to hit.
Also includes inhands.

gibbe instagib CTF pls

### CODE REVIEW

**Comment by Aloraydrel** [comment]
finally :+1:

**Comment by PKPenguin321** [comment]
all we need is unreal's teleport disc gun thing

**Comment by freerealestate** [comment]
>A rifle based on the Unreal series 'instagib' mode, infinite ammo and gibs anything it manages to hit.
Also includes inhands.

File: code/modules/projectiles/ammunition/energy.dm [hunk 1]
```diff
@@ -166,3 +166,11 @@ obj/item/ammo_casing/energy/net
...
 	projectile_type = /obj/item/projectile/energy/trap
 	select_name = "snare"
+
+/obj/item/ammo_casing/energy/instakill
+	projectile_type = /obj/item/projectile/beam/instakill
+	e_cost = 0
+	select_name = "DESTROY"
+
+/obj/item/ammo_casing/energy/instakill/blue
+	projectile_type = /obj/item/projectile/beam/instagib/blue
```
Comment (neersighted) [inline code/modules/projectiles/ammunition/energy.dm]: Also make the red one a child?

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -284,3 +284,20 @@
...
+	item_state = "instagib"
+	desc = "A specialized ASMD laser-rifle, capable of flat-out disintegrating most targets in a single hit."
+	ammo_type = list(/obj/item/ammo_casing/energy/instakill)
+	force = 60
+	origin_tech = null
+
+/obj/item/weapon/gun/energy/laser/instakill/blue
+	icon_state = "instagibblue"
+	item_state = "instagibblue"
+	ammo_type = list(/obj/item/ammo_casing/energy/instakill/blue)
```
Comment (neersighted) [inline code/modules/projectiles/guns/energy/special.dm]: Ditto, make the red one a child.

File: code/modules/projectiles/projectile/beams.dm [hunk 3]
```diff
@@ -92,3 +92,19 @@
...
+
+/obj/item/projectile/beam/instagib/blue
+	icon_state = "bluelaser"
+
+/obj/item/projectile/beam/instakill/on_hit(atom/target)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		M.gib()
+		M.visible_message("<span class='danger'>[M] explodes into a shower of gibs!</span>")
```
Comment (neersighted) [inline code/modules/projectiles/projectile/beams.dm]: Sounds effects would be great. MULTIKILL!

**Comment by Shadowlight213** [comment]
Inb4 add to Rnd. honk honk

File: code/modules/projectiles/projectile/beams.dm (hunk 3 above)
Comment (hornygranny) [inline code/modules/projectiles/projectile/beams.dm]: the visible message doesn't work because it's after gib, which removes M

**Comment by hornygranny** [comment]
it needs a better projectile

**Comment by WJohn** [comment]
:-1: on the sprites. Lacks an outline, and looks more like a single blob of metal rather than segmented parts.

**Comment by bgobandit** [comment]
is this blacklisted from summon guns

**Comment by MMMiracles** [comment]
@bgobandit summon guns uses a white list of approved guns, so as hilarious as it'd be for an instagib rifle to spawn during summon guns, it won't happen.

**Comment by freerealestate** [comment]
this is overpowered and imo shouldn't be in the game, even locked behind admin favouritism/self-spawning

**Comment by MMMiracles** [comment]
@freerealestate 
I made this solely because Kor said he'd make a instagib variant of CTF if I made the gun. 

Also who in their right mind would spawn this in for anyone during a regular round. 

>inb4 someone does a tc trade for it with an unknowing admin

**Comment by AlexanderUlanH** [comment]
>unknowing admin

"Instagib rifle?  What's the worst that could happen?"

**Comment by WJohn** [comment]
The sprites are slightly less bad, but could still use a lot more detail. Out of curiosity, which instagib rifle are you basing this off of? I seem to recall the unreal games having an awful lot of instakill weapons.

**Comment by MMMiracles** [comment]
A mix between UT3 and UT2K4 was my goal.

**Comment by GunHog** [comment]
Add the Redeemer next. It is a remote controlled NUKE.

**Pull request merged**